### DISCUSSION TITLE
Beamer: Render classoption: "handout, presentation" simultaneously

### DISCUSSION THREAD

**Discussion by math-mcshane** [quarto-dev/quarto-cli#7218] [Q&A]
### Description

Context: I teach using "animated" Beamer slides (e..g, `classoption: "presentation"`) but share the "static" slides with students (e.g., `classoption: "handout"`). They prefer writing on the final slide rather than the intermediate slides. As of now, as far as I can tell, it's not possible to render both simultaneously from directly within Beamer (although there may be some workarounds with `.yml` files -- I haven't taken too close of a look at those). 

It seems like it might be as straightforward as: 

1. Allowing `classoption: "handout, presentation"`
2. When this parameter is used, render the slides as usual to `"presentation"` format; save as `[.qmd_filename]_presentation.pdf`
3. Then, create a copy of `[.qmd_filename]_presentation.pdf` to `[.qmd_filename]_handout.pdf` and delete all the intermediate slides. I suspect this would be the preferred option over rendering twice, as some slides could take a while to render. 

Apologies if there's already a way to do something like this. I'm a recent Quarto convert from R Markdown. 

## Here's a minimal working example (just paste into .qmd) with slides animated differently and one static content slide. 

```
---
title: "MWE"
author: "Ryan McShane"
format: 
  beamer:
    header-includes: |
      \setbeamertemplate{footline}{\insertframenumber{} / \inserttotalframenumber}
classoption: "presentation"
# classoption: "handout"
---

## Slide with "animations" the LaTeX way

I think,

\pause

therefore I am.

$\square$

## Slide with "animated" bullets  the Quarto way

::: {.incremental}

- Eat spaghetti
- Drink wine

:::

Content before pause

. . .

content after pause



## Static Slide

Hello, it me

```

**Comment by cderv** [comment]
> Allowing classoption: "handout, presentation"

`classoption` can currently take a list of values and they will be use in template to set in document class.
https://github.com/quarto-dev/quarto-cli/blob/b6f50390956bec686d06a01430e0a87e39de92c3/src/resources/formats/pdf/pandoc/latex.template#L27-L29

we can't make a behavior specific this way at this level to know what a character separated by a comma means. 

>  As of now, as far as I can tell, it's not possible to render both simultaneously from directly within Beamer (although there may be some workarounds with .yml files -- I haven't taken too close of a look at those).

This would be the perfect usage for profile (https://quarto.org/docs/projects/profiles.html) as you can set different format configuration based on each profile, and render your slides twice, once for each profile. 

Also there is a hidden trick with `format` keys: You can add modifiers to it to use same format with different options
So you could do 
````yaml
format: 
  beamer-presentation:
    classoption: "presentation"
  beamer-handout: 
    classoption: "handout"
````

Any modifier which is not Pandoc extensions works, and will be used in the output file name. So this will create `[.qmd_filename]-presentation.pdf` and `[.qmd_filename]-handout.pdf`

Also use `include-in-header`  as this is the Quarto way : https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes 

````markdown
---
title: "MWE"
author: "Ryan McShane"
format: 
  beamer-prespresentation:
    classoption: "presentation"
  beamer-handout: 
    classoption: "handout"
include-in-header: 
  text: |
    \setbeamertemplate{footline}{\insertframenumber{} / \inserttotalframenumber}
---

## Slide with "animations" the LaTeX way

I think,

\pause

therefore I am.

$\square$

## Slide with "animated" bullets  the Quarto way

::: {.incremental}

- Eat spaghetti
- Drink wine

:::

Content before pause

. . .

content after pause



## Static Slide

Hello, it me

````

Hope it helps

**Comment by math-mcshane** [comment]
Hi @cderv thanks for the info! As always, you are very helpful (I have learned a lot from your comments to others on github over the past few years). 

If I read correctly, the format keys approach automatically generates different names, although still requires rendering twice (?). This does reduce the hassle a bit! I'll at least start doing that right away. OK, helpful note to self: the hidden format option that is at the top is the default on first render, so put `handout` first. 

Also, I don't know how that `header-includes` slipped into my `MWE.qmd`! I've been trying so hard to use `include-in-header`!

There's a lot going on on the profiles page; I'll have to read more deeply. But are you saying that I could _render some number of formats >1, so as long as I have it defined in a_ `.yml`? I haven't yet branched out to rendering RMDs or QMDs from a separate file.

**Reply by mcanouil** [reply]
If you use the format trick, you can render to all formats.
If I recall correctly it should be `quarto render --to all` (not in a setting right now where I can double check this).

PS: changing to Q&A.

**Reply by cderv** [reply]
Yes this is it `quarto render <input> --to all`. This will render to all mentioned format with one call to render !
